muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/fieldio/FileAnswerUtils.java

```diff
@@ -0,0 +1,151 @@
+package fi.otavanopisto.muikku.plugins.workspace.fieldio;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import fi.otavanopisto.muikku.dao.base.SystemSettingDAO;
+import fi.otavanopisto.muikku.model.base.SystemSetting;
+import fi.otavanopisto.muikku.plugins.workspace.dao.WorkspaceMaterialFileFieldAnswerFileDAO;
+import fi.otavanopisto.muikku.plugins.workspace.model.WorkspaceMaterialFileFieldAnswer;
+import fi.otavanopisto.muikku.plugins.workspace.model.WorkspaceMaterialFileFieldAnswerFile;
+
+public class FileAnswerUtils {
+  
+  public static final String SETTING_ENABLED = "fileanswers.storageEnabled";
+  public static final String SETTING_FILEFOLDER = "fileanswers.storageFolder";
+  public static final String SETTING_LASTID = "fileanswers.lastId";
+
+  @Inject
+  private Logger logger;
+
+  @Inject
+  private SystemSettingDAO systemSettingDAO;
+  
+  @Inject
+  private WorkspaceMaterialFileFieldAnswerFileDAO workspaceMaterialFileFieldAnswerFileDAO;
+  
+  public Long getLastEntityId() {
+    SystemSetting idSetting = systemSettingDAO.findByKey(SETTING_LASTID);
+    return idSetting == null ? 0L : Long.valueOf(idSetting.getValue());
+  }
+  
+  public void setLastEntityId(Long id) {
+    SystemSetting idSetting = systemSettingDAO.findByKey(SETTING_LASTID);
+    if (idSetting == null) {
+      systemSettingDAO.create(SETTING_LASTID, id.toString());
+    }
+    else {
+      systemSettingDAO.update(idSetting, id.toString());
+    }
+  }
+  
+  public boolean isFileSystemStorageEnabled() {
+    SystemSetting enabledSetting = systemSettingDAO.findByKey(SETTING_ENABLED);
+    return enabledSetting != null && Boolean.valueOf(enabledSetting.getValue());
+  }
+  
+  public byte[] getFileContent(Long userEntityId, String fileId) throws IOException {
+    if (userEntityId == null || fileId == null) {
+      logger.severe(String.format("Invalid fetch parameters: file %s of user %d", fileId, userEntityId));
+      return new byte[0];
```

Comment:
Might warrant throwing exceptions like https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html

Reply:
Runtime exceptions in EJBs aren't exactly encouraged but given that these occasions would indicate a serious coding error, I suppose it doesn't matter as much.